packages/binding-coap/src/coap-server.ts

```diff
@@ -337,387 +335,478 @@ export default class CoapServer implements ProtocolServer {
-                                        debug(
-                                            `CoapServer on port ${this.getPort()} failed '${segments[3]}' subscription`
-                                        );
-                                        res.code = "5.00";
-                                        res.end();
-                                    }
-                                };
-
-                                thing
-                                    .handleSubscribeEvent(segments[3], listener, options)
-                                    .then(() => res.end())
-                                    .catch(() => res.end());
-                                res.on("finish", () => {
-                                    debug(
-                                        `CoapServer on port ${this.getPort()} ends '${
-                                            segments[3]
-                                        }' observation from ${Helpers.toUriLiteral(req.rsinfo.address)}:${
-                                            req.rsinfo.port
-                                        }`
-                                    );
-                                    thing.handleUnsubscribeEvent(segments[3], listener, options);
-                                });
-                            } else if (req.headers.Observe > 0) {
-                                debug(
-                                    `CoapServer on port ${this.getPort()} sends '${
-                                        segments[3]
-                                    }' response to ${Helpers.toUriLiteral(req.rsinfo.address)}:${req.rsinfo.port}`
-                                );
-                                // node-coap does not support GET cancellation
-                                res.code = "5.01";
-                                res.end("node-coap issue: no GET cancellation, send RST");
-                            } else {
-                                debug(
-                                    `CoapServer on port ${this.getPort()} rejects '${
-                                        segments[3]
-                                    }' read from ${Helpers.toUriLiteral(req.rsinfo.address)}:${req.rsinfo.port}`
-                                );
-                                res.code = "4.00";
-                                res.end("No Observe Option");
-                            }
-                        } else {
-                            res.code = "4.05";
-                            res.end("Method Not Allowed");
-                        }
-                        // resource found and response sent
-                        return;
-                    } // Event exists?
+        }
+
+        const observe = req.headers.Observe as number;
+
+        if (observe == null) {
+            debug(
+                `CoapServer on port ${this.getPort()} rejects '${affordanceKey}' read from ${Helpers.toUriLiteral(
```

Comment:
I think the debug statement might convey that it deals with an event

Reply:
Would this be a better alternative?

```suggestion
                `CoapServer on port ${this.getPort()} rejects '${affordanceKey}' event subscription from ${Helpers.toUriLiteral(
```

Reply:
Took the freedom of already adding this suggestion to the PR as part of 5d0f086c474935df016f5c0b5031e1798ace0328, hope that was okay :)